### PR TITLE
[elfio] Add new vcpkg port elfio

### LINKED ISSUE #14313: [New Port Request] elfio
Library name: elfio

Library description: ELFIO is a header-only C++ library intended for reading and generating files in the ELF binary format. 

Source repository URL: https://github.com/serge1/ELFIO

### PR DESCRIPTION
**Pull request by schultetwin1**
Adds a new new port the elfio library

Fixes #14313

### CODE REVIEW

**Comment by Link1J** [comment]
I wrote one for this lib myself, as I planned to use it. Funny how someone already did it.
I just have one thing to ask, Is there any reason you used a usage file over using `vcpkg_fixup_cmake_targets` in the portfile?

**Comment by schultetwin1** [comment]
> Is there any reason you used a usage file over using vcpkg_fixup_cmake_targets in the portfile?

Only that I did not know it existed :) Thanks for the pointer!

File: ports/elfio/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    SOURCE_PATH ${SOURCE_PATH}
+    PREFER_NINJA)
+
+vcpkg_install_cmake()
+
+vcpkg_fixup_cmake_targets(
+    CONFIG_PATH
+        share/${PORT}/cmake
+    TARGET_PATH
+        share/${PORT})
```
Comment (NancyLi1013) [inline ports/elfio/portfile.cmake:17]:
```suggestion
vcpkg_fixup_cmake_targets(CONFIG_PATH share/${PORT}/cmake TARGET_PATH share/${PORT})
```

File: ports/elfio/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug")
+file(INSTALL ${SOURCE_PATH}/COPYING DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
+
```
Comment (NancyLi1013) [inline ports/elfio/portfile.cmake:22]:
```suggestion
```

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Review by NancyLi1013 — APPROVED** [review]

**Comment by NancyLi1013** [comment]
LGTM, thanks for your PR.

**Comment by BillyONeal** [comment]
Thanks for your help!

**Comment by schultetwin1** [comment]
> Thanks for your help!

Thanks for your changes as well!

**Pull request merged**